lib/modules/social/rest_api.php

```diff
@@ -81,51 +85,200 @@ public function get_contributor_services($request)
     }
 }
 
-class REST_API_V2
-{
-    const api_namespace = 'podlove/v2';
-    const api_base = 'social';
 
-    // todo: delete
-    // todo: create
-    // todo: update
+class WP_REST_PodloveContributorService_Controller extends WP_REST_Controller
+{
+    public function __construct()
+    {
+        $this->namespace = "podlove/v2";
+        $this->rest_base = "social";
+    }
 
     public function register_routes()
     {
-        register_rest_route(self::api_namespace, self::api_base.'/services', [
+        register_rest_route($this->namespace, $this->rest_base.'/services', [
             [
                 'methods' => \WP_REST_Server::READABLE,
-                'callback' => [$this, 'get_services'],
+                'callback' => [$this, 'get_items'],
                 'args' => [
                     'category' => [
-                        'description' => __('category: social, donation, internal'),
+                        'description' => __('category: social, donation, internal', 'podlove-podcasting-plugin-for-wordpress'),
                         'type' => 'string',
                     ],
                 ],
-                'permission_callback' => '__return_true',
+                'permission_callback' => [$this, 'get_item_permissions_check'],
             ],
         ]);
 
-        register_rest_route(self::api_namespace, self::api_base.'/services/contributor/(?P<id>[\d]+)', [
+        register_rest_route($this->namespace, $this->rest_base.'/services/(?P<id>[\d]+)', [
             [
                 'methods' => \WP_REST_Server::READABLE,
-                'callback' => [$this, 'get_contributor_services'],
+                'callback' => [$this, 'get_item'],
+                'permission_callback' => [$this, 'get_item_permissions_check'],
+            ],
+        ]);
+
+        register_rest_route($this->namespace, $this->rest_base.'/contributors/(?P<id>[\d]+)', [
+            [
+                'methods' => \WP_REST_Server::READABLE,
+                'callback' => [$this, 'get_items_contributor_services'],
```

Comment:
Auch hier in der ganzen Datei: kann das `item` in den Methodennamen ggf. überall weg?

Reply:
Ich habe jetzt das item aus den Methoden entfernt. Bei get_item_permission_check (u.ä.) habe ich die Namen beibehalten, da diese auch in der Basis-Klasse so lauten.